src/components/smart_objects/src/smart_object.cc

```diff
@@ -723,16 +723,28 @@ void SmartObject::duplicate(const SmartObject& OtherObject) {
 void SmartObject::cleanup_data() {
   switch (m_type) {
     case SmartType_String:
-      delete m_data.str_value;
+      if (m_data.str_value) {
```

Comment:
@BrandonHe 
1.  C++ standard allows NULL-pointer deletion.
Please see C++98 standard 5.3.5 Delete paragraph 2.
> In either alternative, if the value of the operand of `delete` is null pointer the operation has no effect

2.
> And Also, there is a very small possibility that delete fails, that may cause memory leak

Could you please let me know in which way `if-condition` helps to prevent memory leak?

>Third, this is to avoid of dangling pointer.

To avoid dangled pointer you are using `NULL` assignment which is good.

So, all I want is to avoid redundant `if-condition`